modules/idlgen/core/src/test/scala/higherkindness/mu/rpc/idlgen/ProtoSrcGenTests.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017-2019 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package higherkindness.mu.rpc.idlgen
+
+import java.io.File
+import higherkindness.mu.rpc.idlgen.proto.ProtoSrcGenerator
+import higherkindness.mu.rpc.common.RpcBaseTestSuite
+
+class ProtoSrcGenTests extends RpcBaseTestSuite {
+
+  val module: String  = new java.io.File(".").getCanonicalPath
+  val protoFile: File = new File(module + "/src/test/resources/proto/book.proto")
+
+  "Proto Scala Generator" should {
+
+    "generate correct Scala classes" in {
+
+      val result: Option[(String, Seq[String])] =
+        ProtoSrcGenerator
+          .generateFrom(inputFile = protoFile, serializationType = "", options = "")
+          .map(t => (t._1, t._2.map(_.clean)))
+
+      result shouldBe Some(("com/proto/book.scala", Seq(expectation.clean)))
+    }
+  }
+
+  val expectation =
+    """package com.proto
+      |
```

Comment:
Out of curiosity, this code won't compile because we are not importing `fs2.Stream`, right?

Reply:
Yes, exactly, it doesn't compile because of: `Stream`, `@message` and `@service`.
This is why I replaced the `> publishLocal` by `> srcGen` in the scripted test, since I just wanted to prove that the Scala code has been generated.

Reply:
@juanpedromoreno After:
- Injecting some imports for dealing with `@message`, `@service`, `fs2.Stream`, `:+:` and `CNil`
- Adding some transformations to replace `Cop[A :: B :: TNil]` into `A :+: B :+: CNil`
- Adding the proper dependencies in the scripted test

The right compilation of the generates Scala code is being proved by `scripted sbt-mu-idlgen/srcProtoGenFromDirs`

Done at:
- https://github.com/higherkindness/mu/pull/577/commits/58a34329032b5af9a42834fccaa414a9be1efde5
- https://github.com/higherkindness/mu/pull/577/commits/1281ceef9eab991532923b9d91fc054082594ad9